sqlitecluster/SQLite.cpp

```diff
@@ -819,9 +821,27 @@ int SQLite::commit(const string& description, function<void()>* preCheckpointCal
             if (_sharedData.outstandingFramesToCheckpoint) {
                 auto start = STimeNow();
                 int framesCheckpointed = 0;
-                sqlite3_wal_checkpoint_v2(_db, 0, SQLITE_CHECKPOINT_PASSIVE, NULL, &framesCheckpointed);
+
+                // We default to PASSIVE checkpoint everywhere as that has been the value proven to work fine for many years.
+                if (_checkpointMode != "PASSIVE") {
+                    int checkpointMode = SQLITE_CHECKPOINT_PASSIVE;
```

Comment:
I would just make `_checkpointMode` an `int` instead of doing this string->int translation every time.

I would add a function like:
```
static int SQLite::getCheckpointModeFromString(const string& checkpointModeString) {
    if (checkpointModeString == "PASSIVE") return SQLITE_CHECKPOINT_PASSIVE;
    if (checkpointModeString == "FULL") return SQLITE_CHECKPOINT_FULL;
    if (checkpointModeString == "RESTART") return SQLITE_CHECKPOINT_RESTART;
    if (checkpointModeString == "TRUNCATE") return SQLITE_CHECKPOINT_TRUNCATE;
    SERROR("Invalid checkpoint type: " << checkpointModeString);
}
```

I think this makes everything else easier.

You won't need to check `args["-checkpointMode"]` in `BedrockServer`.

When you initialize `_checkpointMode` here: https://github.com/Expensify/Bedrock/pull/2009/files#diff-6cca2e6a219aa7f2d6c430b05478aa480eeaafd46c51bbe7e4e43d9ac7a64ce9R235

You can just do:
```
_checkpointMode(getCheckpointModeFromString(checkpointMode))
```

You don't need this whole block that I'm commenting in now, you just do:
```
sqlite3_wal_checkpoint_v2(_db, 0, checkpointMode, NULL, &framesCheckpointed);
if (_checkpointMode != SQLITE_CHECKPOINT_PASSIVE) {
    sqlite3_busy_timeout(_db, 120'000);
}
```

And that's this whole section.

Reply:
Updated